File: php/survival.R
# Args <- commandArgs()
# gene<- unlist(strsplit(Args[6],split='_'))
# cancer<- unlist(strsplit(Args[7],split=','))
# signature<-Args[8]
# normalMed<-Args[9]
# normalGene<-Args[10]
# corMed<-Args[11]
# mergedatasets<-Args[12]
# datatype<-Args[13]
# 
# maintitle=paste(gene,cancer,signature,normalMed,normalGene,corMed,mergedatasets,datatype,sep="-")
# maintitle2=paste(gene,cancer,signature,normalMed,normalGene,corMed,mergedatasets,datatype,sep="\n")
maintitle='abc'
png(file=paste("./img/",maintitle,".png",sep=""))
background<-rgb(255, 246, 237,max=255)
par(bg=background)
plot(1:20)
text(10,10, maintitle)
dev.off()



cat (maintitle)

File: php/survival.php
<?php 
header("Content-Type:application/json;charset=UTF8");
$gene=$_GET['gene']; 
#$cancer=$_GET['cancer'];
$signature=$_GET['signature'];
$normalMed=$_GET['normalMed'];
$normalGene=$_GET['normalGene'];
$corMed=$_GET['corMed'];
$mergedatasets=$_GET['mergedatasets'];
$datatype=$_GET['datatype'];

$zhiling="sudo Rscript survival.R $gene  $normalMed $normalGene $corMed $mergedatasets $datatype";

exec("Rscript survival.R $zhiling",$output, $status);

//执行成功都是返回0
echo json_encode(array(
   "gene" => $gene,
   "signature" =>$signature,
   "normalMed" =>$normalMed,
   "normalGene" =>$normalGene,
   "corMed" =>$corMed,
   "mergedatasets" =>$mergedatasets,
   "datatype" =>$datatype,
   "zhiling" =>$zhiling,
   "output" => $output,
   "status"=>$status
    ),JSON_UNESCAPED_UNICODE); 

?>